install/kubernetes/helm/istio/charts/pilot/templates/deployment.yaml

```diff
@@ -114,6 +114,10 @@ spec:
           - name: PILOT_ENABLE_PROTOCOL_SNIFFING
             value: "{{ .Values.enableProtocolSniffing }}"
 {{- end }}
+{{ - if .Values.protocolDetectionTimeout }}
+          - name: PILOT_PROTOCOL_DETECTION_TIMEOUT
+            value: "{{ .Values.protocolDetectionTimeout }}"
```

Comment:
This is exactly what I am suggesting to NOT do. No environment variables. This goes into mesh config